### PR TITLE
JDK-8280738: Minor cleanup for HtmlStyle

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review a simple update to reorder some members of the enum `HtmlStyle`, and to make some minor updates to the comments.

This is to facilitate automated extraction of information into an upcoming new spec for the output from the standard doclet.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8280738](https://bugs.openjdk.java.net/browse/JDK-8280738): Minor cleanup for HtmlStyle

### Reviewers
 * [Hannes Wallnöfer](https://openjdk.java.net/census#hannesw) (@hns - **Reviewer**) ⚠️ Review applies to df8c810eba49c834da05bca0352d3333751d135f

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7235/head:pull/7235` \
`$ git checkout pull/7235`

Update a local copy of the PR: \
`$ git checkout pull/7235` \
`$ git pull https://git.openjdk.java.net/jdk pull/7235/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7235`

View PR using the GUI difftool: \
`$ git pr show -t 7235`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7235.diff">https://git.openjdk.java.net/jdk/pull/7235.diff</a>

</details>

### CODE REVIEW

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/markup/HtmlStyle.java [hunk 1]
```diff
@@ -48,20 +48,14 @@
  * or {@link javax.lang.model.element.Element "language model elements"}.
  * The usage is made explicit when it is not clear from the surrounding context.
  *
+ * @apiNote
+ * The stylized use of {@code editor-fold} comments and line comments (beginning {@code //}
+ * is to support extracting details of declarations with external tools.  Edit with care!
```
Comment (hns) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/markup/HtmlStyle.java:53]: Missing closing parenthesis

**Review by hns — APPROVED** [review]
Looks good. The `@apiNote` regarding editor-folds could be improved (missing parenthesis).

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/markup/HtmlStyle.java (hunk 1 above)
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/markup/HtmlStyle.java:53]: Noted

**Pull request closed without merging**